chips/apollo3/src/stimer.rs

```diff
@@ -115,45 +119,87 @@ impl<'a> STimer<'_> {
         // Clear interrupt
         regs.stmintclr.modify(STMINT::COMPAREA::SET);
 
-        self.client.map(|client| client.fired());
+        self.client.map(|client| client.alarm());
     }
 }
 
-impl<'a> hil::time::Alarm<'a> for STimer<'a> {
-    fn set_client(&self, client: &'a dyn hil::time::AlarmClient) {
-        self.client.set(client);
+impl Time for STimer<'_> {
+    type Frequency = Freq16KHz;
+    type Ticks = Ticks32;
+
+    fn now(&self) -> Ticks32 {
+        Ticks32::from(self.registers.sttmr.get())
     }
+}
 
-    fn set_alarm(&self, tics: u32) {
-        let regs = self.registers;
+impl<'a> Counter<'a> for STimer<'a> {
+    fn set_overflow_client(&'a self, _client: &'a dyn OverflowClient) {
+        //self.overflow_client.set(client);
+    }
 
+    fn start(&self) -> ReturnCode {
         // Set the clock source
-        regs.stcfg.write(STCFG::CLKSEL::XTAL_DIV2);
```

Comment:
This is no longer correctly called.

`STCFG::CLKSEL::XTAL_DIV2` is set only in `Counter` `start()`.

If not using the `Counter` but only using the `Alarm` this is never set.

Adding `self.registers.stcfg.write(STCFG::CLKSEL::XTAL_DIV2);` to `set_alarm()` fixes this.

Reply:
Does it make sense to select the clock source every time an alarm is set? This is a configuration action so should in theory only be done whenever the clock needs to be changed. Other alarms have a configure call that does this. Or does the apollo3 require you reselect the clock every time you set the compare register?

Reply:
I updated the boot sequence to start the counter.

Reply:
Works